### ISSUE TITLE
Per-language terminology database

### ISSUE THREAD

**Issue by nmlgc** [thpatch/thpatch#4]
On the existing Touhou Patch Center, we've always told translators to use the [Portal pages](https://thpatch.net/w/index.php?title=Special%3AAllPages&from=&to=&namespace=150) for discussing the translation guidelines for their respective language and presenting them in any format. However, since MediaWiki markup is far from being easy to use, many translation teams instead chose to keep their guidelines off-site in a more comfortable format, e.g. [a Google spreadsheet](https://docs.google.com/spreadsheets/d/1xf_g7eT3LHymMsuUqTzGFqGhHKsaUaOB6itYvs1uKEc) or a [thread in the team's existing forum](http://www.touhou-online.net/communaute/forum/4-Projets/6210-important,-complexe-traduction-des-jeux-touhou-%E2%80%93-harmonisation).

While we probably shouldn't add either a forum or a spreadsheet system to the new Touhou Patch Center, we could provide at least a database for certain recurring Japanese terms and their recommended translation in any language. This would come with a number of advantages:
- Updating the list of original Japanese terms could be done by the site staff as part of adding translation support for a new game, instead of expecting each team to do this on their own. (This is where having staff who understands Japanese and actually _cares_ about the content certainly helps.)
- In the translation interface, we could parse the original text, look up any of the terms contained in it in the database, and display their recommended translation above the text box. (Thankfully, this is pretty trivial with Japanese and doesn't require any natural language parsing.)
- Maybe we could even refuse to save a translation if any of the terms in the original text aren't translated according to the database and the translator hasn't clicked some sort of _"I know what I'm doing"_ check box. (…OK, this _does_ require natural language parsing.)
- On the other hand, whenever a term in the _database_ is changed to a different translation, we could directly display all translation units that are affected by the change and that now need to be retranslated. (Also requires natural language parsing.)

**Comment by brliron** [comment]
I don't think we really need natural language parsing for the 3rd and 4th points. Some easy substring search may be enough.
For example, let's take https://thpatch.net/wiki/Th11/Reimu_and_Yukari%27s_Scenario , stage 4, #0@218. In japanese, it is :
申し遅れました
私はさとり、この地霊殿の主です
And the english version is :
Oh, excuse me. I'm Satori,
the master of the Palace of the Earth Spirits.
The japanese version contains 地霊殿 one time. So we count how many times "Palace of the Earth Spirits" appears in the english version. Here, we see it 1 time, so it's okay. If we saw it 0 or 2 times, we would display the warning.
And for the 4th point, when someone changes a term in the database, we replace each occurrence of <old translated term> with <new translated term>. And we can display each change to the user, so if this replacing made some shit, the user can fix it.

**Comment by IgoSplashman** [comment]
Staff only: Agreed.
Recommended translation and notification when a term gets changed: Muchly approved.
Enforcement: No, or else we'll lose the human element if we force everyone to limit themselves to only certain terms. Like in th14 you can only write "social upheaval" so many times until the characters start to sound like parrots. I firstly created the table as a guidline, place for discussion and a quick reference tool.

However, we could set an enforce flag for only specific terms, especially to keep the spelling in check. For example, is it "Schreinmädchen" or "Schrein-Mädchen", "Gensokyo" or "Gensoukyou"?
